packages/shorebird_cli/lib/src/artifact_builder.dart

```diff
@@ -0,0 +1,127 @@
+import 'dart:io';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/os/operating_system_interface.dart';
+import 'package:shorebird_cli/src/platform/platform.dart';
+import 'package:shorebird_cli/src/shorebird_android_artifacts.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_process.dart';
+
+/// Used to wrap code that invokes `flutter build` with Shorebird's fork of
+/// Flutter.
+typedef ShorebirdBuildCommand = Future<void> Function();
+
+/// {@template build_exception}
+/// Thrown when a build fails.
+/// {@endtemplate}
+class BuildException implements Exception {
+  /// {@macro build_exception}
+  BuildException(this.message);
+
+  /// Information about the build failure.
+  final String message;
+}
+
+final artifactBuilderRef = create(ArtifactBuilder.new);
+
+ArtifactBuilder get artifactBuilder => read(artifactBuilderRef);
+
+class ArtifactBuilder {
```

Comment:
Would be nice to add some docs to this class and all public APIs

Reply:
Done